### PR TITLE
Added support for kotlin power assert

### LINKED ISSUE #4044: Power-assert for better failure message formatting
[Power assert](https://kotlinlang.org/docs/power-assert.html) is a Kotlin compiler plugin for printing nicely diagrammed failure messages in tests. From the lib's readme:

Given following code:
```kotlin
val hello = "Hello"
assert(hello.length == "World".substring(1, 4).length)
```

Normally the assertion message would look like:
```kotlin
java.lang.AssertionError: Assertion failed
	at <stacktrace>
```

With [kotlin-power-assert](https://github.com/bnorm/kotlin-power-assert/) included, the error message for the previous example will be transformed:
```kotlin
java.lang.AssertionError: Incorrect length
assert(hello.length == "World".substring(1, 4).length)
       |     |      |          |               |
       |     |      |          |               3
       |     |      |          orl
       |     |      false
       |     5
       Hello
	at <stacktrace>
```

Use case in relation to kotest: https://github.com/bnorm/kotlin-power-assert/issues/65#issuecomment-1582065946

YT talk: https://www.youtube.com/live/G8pS1FzZlhk?si=Wt-QlGxxIuX83ZOh&t=17600

### PR DESCRIPTION
**Pull request by sksamuel**
Will require https://github.com/kotest/kotest/pull/4248

Closes https://github.com/kotest/kotest/issues/4044

### CODE REVIEW

File: kotest-assertions/kotest-assertions-core/src/jvmMain/kotlin/io/kotest/matchers/date/zoneddatetime.kt [hunk 1]
```diff
@@ -14,9 +14,8 @@ fun beInTodayZDT() = object : Matcher<ZonedDateTime> {
       return MatcherResult(
          passed,
          { "$value should be today" },
-         {
-            "$value should not be today"
-         })
+         { "$value should not be today" }
```
Comment (AlexCue987) [inline kotest-assertions/kotest-assertions-core/src/jvmMain/kotlin/io/kotest/matchers/date/zoneddatetime.kt:17]: Is this change related to power assert?

Comment (sksamuel) [reply kotest-assertions/kotest-assertions-core/src/jvmMain/kotlin/io/kotest/matchers/date/zoneddatetime.kt:17]: Just clean up

**Comment by AlexCue987** [comment]
this is experimental. are we planning to let users opt in or opt out?

**Comment by sksamuel** [comment]
> this is experimental. are we planning to let users opt in or opt out?

users have to add the plugin themselves to their gradle script to enable this.
What this PR does is support it if they decide to add that plugin.

**Review by AlexCue987 — APPROVED** [review]
we need to document it somewhere in another PR
